### PR TITLE
Google Workspace analytic content update

### PR DESCRIPTION
**Pull request by spsocprime**
Required items, please complete
   
   Change(s):
   - Updated Analytic rules and Hunting queries
         > added pre-filtering conditions to check if a field is not empty `isnotempty(field)`

   Version Updated:
   - Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by vmanojreddy** [comment]
@aprakash13 Please review Analytic rules and Hunting queries. thanks!!!

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Please remove the isnotempty check from all the queries where we do a match right after. It makes the queries take longer. E.g. query

| where isnotempty(EventMessage) 
| where EventMessage =~ 'grant_admin_privilege'

**Comment by spsocprime** [comment]
> Please remove the isnotempty check from all the queries where we do a match right after. It makes the queries take longer. E.g. query
> 
> | where isnotempty(EventMessage) | where EventMessage =~ 'grant_admin_privilege'

fixed

**Comment by vmanojreddy** [comment]
@aprakash13  please check and approve the request changes. thanks!!!

**Comment by vmanojreddy** [comment]
@aprakash13 please check and approve the request changes. thanks!!!

File: Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceDocumentSharedPublicily.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ relevantTechniques:
 query: |
   GWorkspaceActivityReports
   | where TimeGenerated > ago(24h)
+  | where isnotempty(Visibility)
```
Comment (aprakash13) [inline Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceDocumentSharedPublicily.yaml:19]: isnotempty check is not needed as we are comparing in the next line for the same field.

File: Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceSharedPrivateDocument.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ relevantTechniques:
 query: |
   GWorkspaceActivityReports
   | where TimeGenerated > ago(24h)
+  | where isnotempty(OldVisibility) and isnotempty(Visibility)
```
Comment (aprakash13) [inline Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceSharedPrivateDocument.yaml:19]: isnotempty check is not needed as we are comparing in the next line for the same field.

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Queries generally look good. Provided a small suggestion in relation to the isnotempty check. Please modify these.

File: Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceDocumentSharedPublicily.yaml (hunk 1 above)
Comment (spsocprime) [reply Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceDocumentSharedPublicily.yaml]: fixed

File: Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceSharedPrivateDocument.yaml (hunk 2 above)
Comment (spsocprime) [reply Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceSharedPrivateDocument.yaml]: fixed

**Pull request merged**